packaging/os/yaourt.py

```diff
@@ -0,0 +1,385 @@
+        default: "present"
+        choices: ["present", "absent", "latest"]
+
+    recurse:
+        description:
+            - When removing a package, also remove its dependencies, provided
+              that they are not required by other packages and were not
+              explicitly installed by a user.
+        required: false
+        default: "no"
+        choices: ["yes", "no"]
+        version_added: "1.3"
+
+    force:
+        description:
+            - Force remove package, without any checks.
+        required: false
+        default: "no"
+        choices: ["yes", "no"]
+        version_added: "2.0"
+
+    update_cache:
+        description:
+            - Whether or not to refresh the master package lists. This can be
+              run as part of a package installation or as a separate step.
+        required: false
+        default: "no"
+        choices: ["yes", "no"]
+
+    upgrade:
+        description:
+            - Whether or not to upgrade whole system
+        required: false
+        default: "no"
+        choices: ["yes", "no"]
+        version_added: "2.0"
+        
+    upgrade_aur:
+        description:
+            - Whether or not to upgrade whole system and aur-packages 
+        required: false
+        default: "no"
+        choices: ["yes", "no"]
+        version_added: "2.0"
+        
+    check_kernelupdate:
+        description:
+            - Checks if the kernel has been updated
+        default: "no"
+        choices: ["yes", "no"]
+        version_added: "2.0"
+'''
+
+EXAMPLES = '''
+# Install package foo
+- yaourt: name=foo state=present
+
+# Upgrade package foo
+- yaourt: name=foo state=latest update_cache=yes
+
+# Remove packages foo and bar
+- yaourt: name=foo,bar state=absent
+
+# Recursively remove package baz
+- yaourt: name=baz state=absent recurse=yes
+
+# Run the equivalent of "yaourt -Sy" as a separate step
+- yaourt: update_cache=yes
+
+# Run the equivalent of "yaourt -Su" as a separate step
+- yaourt: upgrade=yes
+
+# Run the equivalent of "yaourt -Sua" as a separate step
+- yaourt: upgrade_aur=yes
+
+# Diff "yaourt -Qi linux(-lts)" to "uname -r" as a separate step
+- yaourt: check_kernelupdate=yes
+
+# Run the equivalent of "yaourt -Rdd", force remove package baz
+- yaourt: name=baz state=absent force=yes
+'''
+
+import json
+import shlex
+import os
+import re
+import sys
+
+YAOURT_PATH = "/usr/bin/yaourt"
+
+def get_version(yaourt_output):
+    """Take yaourt -Qi or yaourt -Si output and get the Version"""
+    lines = yaourt_output.split('\n')
+    for line in lines:
+        if 'Version' in line:
+            return line.split(':')[1].strip()
+    return None
+
+def query_package(module, name, state="present"):
+    """Query the package status in both the local system and the repository. Returns a boolean to indicate if the package is installed, and a second boolean to indicate if the package is up-to-date."""
+    if state == "present":
+        lcmd = "yaourt -Qi %s" % (name)
```

Comment:
relies on $PATH ... even though you have a constant with the full path ..., use the module's  find_bin